app-ui/src/pages/traces/Guardrails.tsx

```diff
@@ -523,6 +287,46 @@ function MutatePolicyModalContent(props) {
       <div className="main">
         {!editMode && (
           <div className="collapsable" style={{ width: "100%" }}>
+            {props.policy?.extra_metadata?.detection_rate && (
+              <div className="banner-note info">
+                <BsInfoCircleFill />
+                <span>
+                  Automatically generated rule (detection rate of{" "}
+                  <code>
+                    {props.policy.extra_metadata.detection_rate * 100}%
+                  </code>
+                  ) . Please review before deployment.
+                </span>
+              </div>
+            )}
+            {props.policy?.extra_metadata?.from_url && (
+              <>
+                <h3>Dataset</h3>
+                <DatasetSelector
+                  initialDatasetName={props.dataset.name}
+                  onSelect={(datasetName: string) => {
+                    // set 'last-picked-dataset' in localStorage. This is used in DeployGuardrail.tsx
+                    // to redirect to the selected dataset, next time the /deploy-guardrail page is loaded
+                    // with a pre-filled guardrail code
+                    localStorage.setItem("last-picked-dataset", datasetName);
+
+                    // replace /u/<user>/<dataset> dataset name in the url
+                    // with the selected dataset name
+                    const url = new URL(window.location.href);
+                    const pathParts = url.pathname.split("/");
+                    const userIndex = pathParts.indexOf("u");
+                    const datasetIndex = userIndex + 2;
+                    if (datasetIndex < pathParts.length) {
+                      pathParts[datasetIndex] = datasetName;
+                      url.pathname = pathParts.join("/");
+                      window.history.pushState({}, "", url.toString());
+                    }
+                    // reload the page to load the new dataset
+                    window.location.reload();
+                  }}
+                />
+              </>
+            )}
```

Comment:
show banner/extra UI in case the about-to-be created guardrail comes from a suggestion or the URL